### ISSUE TITLE
Text Encoder

### ISSUE THREAD

**Issue by jaidevshriram** [threestudio-project/threestudio#113]
Hello, I'm trying to repurpose the stable diffusion guidance code for another usecase and for this, I tried to remove the prompt utils dependency. Specifically, I replaced the following code:

```
 text_embeddings = prompt_utils.get_text_embeddings(
          elevation, azimuth, camera_distances, self.cfg.view_dependent_prompting
)
```
with this:

```
        text_embeddings = self.pipe._encode_prompt(
            prompt, # str
            self.device,
            1,
            True,
            # prompt_embeds=None,
            # negative_prompt_embeds=None,
        )
```

However, I am unable to get good results with this modification. I tested this by trying to generate a 2D image using SDS with the prompt utils text embedding vs the one obtained directly from the pipeline's text encoder. The modification leads to random-looking images/shapes. Could you clarify if this modification makes sense / share how I can modify the text embedding portion?

**Comment by bennyguo** [comment]
Our prompt processor returns [positive text embeddings, negative text embeddings], while the encoding function in diffusers pipeline returns [negative text embeddings, positive text embeddings]. So you should chunk the return value into two parts and swap the two :)

**Comment by jaidevshriram** [comment]
That did fix the issue! Just to confirm, you mean the conditional and unconditional text embedding right? not the prompt and negative prompt.

**Comment by bennyguo** [comment]
Yes you just need to swap the embeddings returned by the _encode_prompt function.

**Comment by jaidevshriram** [comment]
Great! That's what I did - closing now. Thanks for the quick reply!
